Mathlib/Data/Set/Finite.lean

```diff
@@ -1392,10 +1392,10 @@ variable [Preorder α] [Nonempty α] {s : Set α}
 
 theorem infinite_of_forall_exists_gt (h : ∀ a, ∃ b ∈ s, a < b) : s.Infinite := by
   inhabit α
-  set f : ℕ → α := fun n => Nat.recOn n (h default).some fun n a => (h a).some
-  have hf : ∀ n, f n ∈ s := by rintro (_ | _) <;> exact (h _).choose_spec.some
-  refine' infinite_of_injective_forall_mem (strictMono_nat_of_lt_succ fun n => _).Injective hf
-  exact (h _).choose_spec.choose_spec
+  set f : ℕ → α := fun n => Nat.recOn n (h default).choose fun n a => (h a).choose
+  have hf : ∀ n, f n ∈ s := by rintro (_ | _) <;> exact (h _).choose_spec.1
+  exact infinite_of_injective_forall_mem
+    (strictMono_nat_of_lt_succ fun n => (h _).choose_spec.2).injective hf
```

Comment:
Mostly irrelevant golfing. The proof could previously not be proved in one go because elaboration wouldn't play ball. It seems that eliminating the inner `choose` fixes things.